app/javascript/components/users/SignInPage.spec.tsx

```diff
@@ -12,62 +15,154 @@ jest.mock('../../legacy_react/src/lib/api/sign_in');
+
+			const button = result.getByTestId('signInButton');
+			// everytime you try to call the User SignIn API in this test, return a
+			// promise which resolves to {id: 1}
+			mockedWebUserSignIn.postSignIn.mockResolvedValue({id: 1});
+
+			// act puts all of the related React updates for the click event into a
+			// single update. Since fireEvent.click calls some promises, we need to make
+			// the callback a Promise and await on act. If we didn't, our test wouldn't
+			// wait for all the possible React changes to happen at once.
+			await act(async () => {
+				fireEvent.click(button);
+			});
+
+			// was document.location.assign called with redirectUrl?
+			expect(locationAssignSpy).toHaveBeenCalledWith('redirectUrl');
+
+			const error = result.getByTestId('signInPageError');
+			expect(error).toBeEmptyDOMElement();
 		});
 
-		const error = result.getByTestId('signInPageError');
-		expect(error).toBeEmptyDOMElement();
 	});
 
 	it('signIn failed', async () => {
 		expect.hasAssertions();
-		const result = render(<Wrapper><SignInPage/></Wrapper>);
-
-		const button = result.getByTestId('signInButton');
-		// everytime you try to call the User SignIn API in this test, return a
-		// promise which resolves to {id: 1}
-		mockedWebUserSignIn.postSignIn.mockResolvedValue({id: 1});
-
-		// act puts all of the related React updates for the click event into a
-		// single update. Since fireEvent.click calls some promises, we need to make
-		// the callback a Promise and await on act. If we didn't, our test wouldn't
-		// wait for all the possible React changes to happen at once.
-		await act(async () => {
-			fireEvent.click(button);
+
+		locationAssign(async (locationAssignSpy:jest.SpyInstance<void, [url: string]>) => {
+
+			const result = render(<Wrapper><SignInPage redirectUrl={'redirectUrl'}/></Wrapper>);
+
+			const button = result.getByTestId('signInButton');
+			// everytime you try to call the User SignIn API in this test, return a
+			// promise which resolves to {id: 1}
+			mockedWebUserSignIn.postSignIn.mockResolvedValue({id: 1});
+
+			// act puts all of the related React updates for the click event into a
+			// single update. Since fireEvent.click calls some promises, we need to make
+			// the callback a Promise and await on act. If we didn't, our test wouldn't
+			// wait for all the possible React changes to happen at once.
+			await act(async () => {
+				fireEvent.click(button);
+			});
+
+			// was document.location.assign not called?
+			expect(locationAssignSpy).not.toHaveBeenCalledWith('redirectUrl');
+
+			const error = result.getByTestId('signInPageError');
+
+			// Sometimes because of React's rendering cycle, the changes haven't been
+			// made to the HTML by the time we need to expect. waitFor tries some sort
+			// function multiple times until it passes (or 5 seconds have passed)
+			waitFor(() => expect(error).toHaveTextContent("Ermahgerd! We had an error!"));
 		});
```

Comment:
We're wrapping most of our test in a function which sets up a mock for the setting the document location and then provides `locationAssignSpy` which gives us the ability to check whether the `document.location.assign` function was called.